storage/replica_command.go

```diff
@@ -1648,7 +1648,7 @@ func (r *Replica) applyNewLeaseLocked(
 			// in on the Lease struct.
```

Comment:
And it feels slightly cleaner to use `lease.Start` (i.e. the passed-in arg) because I had to look up and make sure that you had already overwritten the state and weren't just using the old lease's Start erroneously.